### PR TITLE
`azurerm_api_management`  add `sku_name` `Consumption_0`

### PR DESCRIPTION
**Pull request by yupwei68**
Fix https://github.com/terraform-providers/terraform-provider-azurerm/issues/6730

### CODE REVIEW

**Comment by yupwei68** [comment]
@mbfrahry  But `Consumption_0` works, and it's in the unit test.

**Comment by mbfrahry** [comment]
Whist the unit test passes, I don't believe an acceptance test will. Do you mind adding an acceptance test that provisions the resource with `Consumption_0` to confirm that one gets provisioned successfully.

File: azurerm/internal/services/apimanagement/validate/api_management_test.go
Comment (katbyte) [inline azurerm/internal/services/apimanagement/validate/api_management_test.go:65]: why are these the same except for the casing?

File: azurerm/internal/services/apimanagement/validate/api_management_test.go
Comment (katbyte) [inline azurerm/internal/services/apimanagement/validate/api_management_test.go:68]: why is this one allin uppercase?

File: azurerm/internal/services/apimanagement/validate/api_management_test.go
Comment (katbyte) [inline azurerm/internal/services/apimanagement/validate/api_management_test.go:12]: Is there a reason we are no longer using the SDK constants?

File: azurerm/internal/services/apimanagement/validate/api_management_test.go
Comment (yupwei68) [reply azurerm/internal/services/apimanagement/validate/api_management_test.go:65]: Because case ignored is accepted for api.

File: azurerm/internal/services/apimanagement/validate/api_management_test.go
Comment (yupwei68) [reply azurerm/internal/services/apimanagement/validate/api_management_test.go:68]: Because case ignored is accepted for api.

File: azurerm/internal/services/apimanagement/validate/api_management_test.go
Comment (yupwei68) [reply azurerm/internal/services/apimanagement/validate/api_management_test.go:12]: Because there is not SDK contants for `Consumption_0`, only constants `Consumption`. As the capacity ranges differ by sku type, I use a regular expression to validate the `sku_name`

**Comment by yupwei68** [comment]
Hi @mbfrahry ,you're right. Changes have been pushed to get `Consumption_0` passed. The tests bellow have passed. 
```
=== RUN   TestAccAzureRMApiManagement_basic
=== PAUSE TestAccAzureRMApiManagement_basic
=== CONT  TestAccAzureRMApiManagement_basic
--- PASS: TestAccAzureRMApiManagement_basic (2534.08s)
=== RUN   TestAccAzureRMApiManagement_requiresImport
=== PAUSE TestAccAzureRMApiManagement_requiresImport
=== CONT  TestAccAzureRMApiManagement_requiresImport
--- PASS: TestAccAzureRMApiManagement_requiresImport (2480.47s)
=== RUN   TestAccAzureRMApiManagement_customProps
=== PAUSE TestAccAzureRMApiManagement_customProps
=== CONT  TestAccAzureRMApiManagement_customProps
--- PASS: TestAccAzureRMApiManagement_customProps (2195.22s)
=== RUN   TestAccAzureRMApiManagement_signInSignUpSettings
=== PAUSE TestAccAzureRMApiManagement_signInSignUpSettings
=== CONT  TestAccAzureRMApiManagement_signInSignUpSettings
--- PASS: TestAccAzureRMApiManagement_signInSignUpSettings (2599.11s)
=== RUN   TestAccAzureRMApiManagement_policy
=== PAUSE TestAccAzureRMApiManagement_policy
=== CONT  TestAccAzureRMApiManagement_policy
--- PASS: TestAccAzureRMApiManagement_policy (2873.39s)
=== RUN   TestAccAzureRMApiManagement_consumption
=== PAUSE TestAccAzureRMApiManagement_consumption
=== CONT  TestAccAzureRMApiManagement_consumption
--- PASS: TestAccAzureRMApiManagement_consumption (177.89s)
=== RUN   TestAccAzureRMApiManagement_identityUserAssigned
=== PAUSE TestAccAzureRMApiManagement_identityUserAssigned
=== CONT  TestAccAzureRMApiManagement_identityUserAssigned
--- PASS: TestAccAzureRMApiManagement_identityUserAssigned (2011.97s)
```

File: azurerm/internal/services/apimanagement/api_management_resource.go
Comment (yupwei68) [reply azurerm/internal/services/apimanagement/api_management_resource.go:694]: Here is a special case. It raises  400 error if we get the `sign In` api and `sign Up` api for a Consumption Sku service. So do we need to set empty list for `sign in` and `sign up` block in consuption sku additionally?
image

File: azurerm/internal/services/apimanagement/api_management_resource.go
Comment (yupwei68) [reply azurerm/internal/services/apimanagement/api_management_resource.go:560]: Here is another special case. This `sign in` and `sign up` resource have not delete api. I suppose they are configuration like resources.

File: azurerm/internal/services/apimanagement/api_management_resource.go
Comment (yupwei68) [reply azurerm/internal/services/apimanagement/api_management_resource.go:553]: The same special case.

**Comment by yupwei68** [comment]
hi @katbyte @tombuildsstuff Except the open comments, other changes have been pushed. Please continue your reviewing.
```
=== RUN   TestAccAzureRMApiManagement_basic
=== PAUSE TestAccAzureRMApiManagement_basic
=== CONT  TestAccAzureRMApiManagement_basic
--- PASS: TestAccAzureRMApiManagement_basic (2109.62s)
=== RUN   TestAccAzureRMApiManagement_requiresImport
=== PAUSE TestAccAzureRMApiManagement_requiresImport
=== CONT  TestAccAzureRMApiManagement_requiresImport
--- PASS: TestAccAzureRMApiManagement_requiresImport (2117.30s)
=== RUN   TestAccAzureRMApiManagement_customProps
=== PAUSE TestAccAzureRMApiManagement_customProps
=== CONT  TestAccAzureRMApiManagement_customProps
--- PASS: TestAccAzureRMApiManagement_customProps (2006.74s)
=== RUN   TestAccAzureRMApiManagement_signInSignUpSettings
=== PAUSE TestAccAzureRMApiManagement_signInSignUpSettings
=== CONT  TestAccAzureRMApiManagement_signInSignUpSettings
--- PASS: TestAccAzureRMApiManagement_signInSignUpSettings (2479.38s)
=== RUN   TestAccAzureRMApiManagement_policy
=== PAUSE TestAccAzureRMApiManagement_policy
=== CONT  TestAccAzureRMApiManagement_policy
--- PASS: TestAccAzureRMApiManagement_policy (2197.78s)
=== RUN   TestAccAzureRMApiManagement_consumption
=== PAUSE TestAccAzureRMApiManagement_consumption
=== CONT  TestAccAzureRMApiManagement_consumption
--- PASS: TestAccAzureRMApiManagement_consumption (182.58s)
=== RUN   TestAccAzureRMApiManagement_identityUserAssigned
=== PAUSE TestAccAzureRMApiManagement_identityUserAssigned
=== CONT  TestAccAzureRMApiManagement_identityUserAssigned
--- PASS: TestAccAzureRMApiManagement_identityUserAssigned (2127.51s)
```

File: azurerm/internal/services/apimanagement/validate/api_management_test.go
Comment (yupwei68) [reply azurerm/internal/services/apimanagement/validate/api_management_test.go:12]: Would you mean to use SDK contants in regular expression?

File: azurerm/internal/services/apimanagement/api_management_resource.go
Comment (tombuildsstuff) [reply azurerm/internal/services/apimanagement/api_management_resource.go:695]: That's fine, we don't have to send them to the API - but we do need to set them into the state (as an empty list) - else this'll show as a diff being `null` vs an empty list `[]`

**Comment by yupwei68** [comment]
@tombuildsstuff corresponding changes have been made. Please continue reviewing.

**Comment by sschmeck** [comment]
Hi @katbyte, is your finding regarding the SDK constant waiting for another integration? :)

**Comment by MireilleMedhat** [comment]
Hello, can somebody please provide updates for this PR? is it planned to be merged any time soon?
Thank you:)

**Comment by sschmeck** [comment]
The integration is blocked until #7603 is merged, which waits for the `v14.2.1` release of [Azure/go-autorest](https://github.com/Azure/go-autorest).

**Comment by AviateX14** [comment]
@sschmeck I believe the dependent change in go-autorest https://github.com/Azure/go-autorest/pull/540 has been merged now, as such, could we prgoress this? I'm hitting my head against https://github.com/terraform-providers/terraform-provider-azurerm/issues/6730#issuecomment-623943489 :(

**Comment by yupwei68** [comment]
@AviateX14 What we depends on is the go-autorest from tombuildstuff' repo, which currently is blocked by go-sdk, as he says. And go-sdk is blocked by some problems released by some service teams. But now we're approaching to the release of these two. I believe I'll start on this PR soon.

**Comment by sven-hoffmann** [comment]
Any news here?

**Comment by yupwei68** [comment]
I've merged the upstream master branch. 
All tests have passed. 
New test:
=== RUN   TestAccAzureRMApiManagement_consumption
=== PAUSE TestAccAzureRMApiManagement_consumption
=== CONT  TestAccAzureRMApiManagement_consumption
--- PASS: TestAccAzureRMApiManagement_consumption (183.75s)

**Review by katbyte — APPROVED** [review]
Thanks @yupwei68 - LGTM 👍

**Comment by katbyte** [comment]
@yupwei68 - tests have passed so this is good to merge one the conflicts are resolved

**Comment by yupwei68** [comment]
Thanks @katbyte . I've merged the latest change. Please continue your review.

**Pull request merged**
